### PR TITLE
add defaultPaperSize to ViewOption

### PR DESCRIPTION
**Pull request by skoji**

### CODE REVIEW

File: src/adapt/expr.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ goog.require('adapt.base');
 /**
  * @typedef {{fontFamily:string, lineHeight:number, margin:number, hyphenate:boolean,
  *   	columnWidth:number, horizontal:boolean, nightMode:boolean, spreadView:boolean,
- *      pageBorder:number, enabledMediaTypes:!Object.<string,boolean>}}
+ *      pageBorder:number, enabledMediaTypes:!Object.<string,boolean>,defaultPaperSize:(Object<number,number>|undefined)}}
```
Comment (kwkbtr) [inline src/adapt/expr.js]: I think it should be `Object<string,number>` rather than `Object<number,number>`.

File: src/vivliostyle/viewer.js [hunk 2]
```diff
@@ -53,6 +53,7 @@ goog.scope(function() {
      * - pageViewMode: Page view mode (singlePage / spread / autoSpread). default: singlePage
      * - zoom: Zoom factor with which pages are displayed. default: 1
      * - fitToScreen: Auto adjust zoom factor to fit the screen. default: false
+     * - defaultPaperSize: Default paper size in mm. Effective when @page size is set to auto. default: undefined (means the windows size is used as paper size).
```
Comment (kwkbtr) [inline src/vivliostyle/viewer.js]: I prefer px values rather than mm, but it might be a matter of taste.

File: src/vivliostyle/page.js [hunk 3]
```diff
@@ -197,13 +197,21 @@ vivliostyle.page.evaluatePageSizeAndBleed = function(pageSizeAndBleed, context)
     var cropOffset = bleed + bleedOffset;
     var width = pageSizeAndBleed.width;
     if (width === adapt.css.fullWidth) {
-        evaluated.pageWidth = (context.pref.spreadView ? Math.floor(context.viewportWidth / 2) - context.pref.pageBorder : context.viewportWidth) - cropOffset * 2;
+        if (context.pref.defaultPaperSize) {
+            evaluated.pageWidth = context.pref.defaultPaperSize.width * context.queryUnitSize("mm", false);;
```
Comment (kwkbtr) [inline src/vivliostyle/page.js]: Duplicate semicolon.

File: src/vivliostyle/viewer.js (hunk 2 above)
Comment (cosmo0920) [reply src/vivliostyle/viewer.js]: It is preferable for me to use `px` unit because previous PR(#334) provided `getPageSizes` API whose unit is `px`.

Comment (cosmo0920) [reply src/vivliostyle/viewer.js]: Or, permit to specify paper size unit is also preferable for me.

File: src/adapt/expr.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ adapt.expr.Preferences;
 adapt.expr.defaultPreferences = function() {
     return {fontFamily:"serif", lineHeight:1.25, margin:8, hyphenate:true, columnWidth:25,
         horizontal:false, nightMode:false, spreadView:false, pageBorder:1,
-        enabledMediaTypes:{"print": true}};
+            enabledMediaTypes:{"print": true}, defaultPaperSize: undefined};
```
Comment (kwkbtr) [inline src/adapt/expr.js]: ESLint complains about indent here.

File: src/adapt/expr.js [hunk 5]
```diff
@@ -45,7 +45,8 @@ adapt.expr.clonePreferences = function(pref) {
     return {fontFamily:pref.fontFamily, lineHeight:pref.lineHeight, margin:pref.margin,
         hyphenate:pref.hyphenate, columnWidth:pref.columnWidth, horizontal:pref.horizontal,
         nightMode:pref.nightMode, spreadView:pref.spreadView, pageBorder:pref.pageBorder,
-        enabledMediaTypes:Object.assign({}, pref.enabledMediaTypes)};
+            enabledMediaTypes:Object.assign({}, pref.enabledMediaTypes),
+            defaultPaperSize: pref.defaultPaperSize? Object.assign({}, pref.defaultPaperSize) : undefined};
```
Comment (kwkbtr) [inline src/adapt/expr.js]: ESLint complains about indent here.

**Comment by kwkbtr** [comment]
LGTM 👍

**Pull request merged**